### DISCUSSION TITLE
pip installing spawns "ninja" processes until box OOM's and dies

### DISCUSSION THREAD

**Discussion by darkpixel** [abetlen/llama-cpp-python#1849] [General]
I'm running Linux Mint on a machine with 128 GB RAM and a decent 16-core processor.
Whenever I try to ```pip install llama-cpp-python``` or use the "pre-built" ```pip install llama-cpp-python --extra-index-url https://abetlen.github.io/llama-cpp-python/whl/120``` it will get to the "Building wheel for llama-cpp-python (pyproject.toml) ..." step and apparently spawn "ninja" processes until the box locks up and becomes unusable and then eventually OOM killer kicks on and knifes all sort of critical system processes and I have to reboot.

This is what the ninja process count looked like for a few seconds:
```
~ $ while true; do ps uaxwww | grep -i python | grep ninja | wc -l; sleep 2; done
998
1075
1152
1228
1302
1379
1456
1528
0
0
^C
~ $ ^C
```

If I'm installing the pre-built llama-cpp-python wheel, shouldn't it NOT be compiling?  Am I missing something?

**Comment by bezbahen0** [comment]
I have the same problem, tried to install an older version v0.2.85, because with the previous version everything was fine, but I had to switch to another device. As a result, I get the same results, there are more processes and my RAM and swap are completely filled, after which the processes are killed

**Reply by bezbahen0** [reply]
I tried to install with cuda support and just CPU, same problems
```
CMAKE_ARGS="-DGGML_CUDA=on" pip install llama-cpp-python
pip install llama-cpp-python
```

**Reply by bezbahen0** [reply]
I solved this problem by switching from ninja==1.11.1 to ninja-1.11.1.2
